src/app/tests/suites/commands/interaction_model/InteractionModel.h

```diff
@@ -174,30 +174,43 @@ class InteractionModelWriter
                               chip::AttributeId attributeId, const T & value,
                               const chip::Optional<uint16_t> & timedInteractionTimeoutMs = chip::NullOptional,
                               const chip::Optional<bool> & suppressResponse              = chip::NullOptional,
-                              const chip::Optional<chip::DataVersion> & dataVersion      = chip::NullOptional)
+                              const chip::Optional<chip::DataVersion> & dataVersion      = chip::NullOptional,
```

Comment:
at what point should we add some sort of builder for `WriteOptions` instead of adding more and more arguments?

This method used to take 8 argumenst (yikes) and now it takes 11 (double yikes), out of which at least 3 are `Optional<uint16_t>`. 

Since these arguments would consume stack anyway when called, lets create a `WriteOptions` class that contains all data instead of us keeping adding arguments

Reply:
I agree... I just don't want to do that in this PR, I would prefer a separate PR just for this task since it may break both `chip-tool` and the whole YAML test suites...

I have opened #19101 for it.